libs/@blockprotocol/graph/src/shared/stdlib/subgraph/edge/entity-type.ts

```diff
@@ -24,27 +22,30 @@ import {
 export const getPropertyTypesReferencedByEntityType = (
   subgraph: Subgraph<boolean>,
   entityTypeId: OntologyTypeVertexId | VersionedUrl,
-): OntologyTypeVertexId[] => {
-  let baseUrl: BaseUrl;
-  let revisionId: OntologyTypeRevisionId;
-
-  if (typeof entityTypeId === "string") {
-    baseUrl = extractBaseUrl(entityTypeId);
-    revisionId = extractVersion(entityTypeId).toString();
-  } else {
-    baseUrl = entityTypeId.baseId;
-    revisionId = entityTypeId.revisionId;
-  }
-
-  const outwardEdges = subgraph.edges[baseUrl]?.[
-    revisionId
-  ] as OntologyOutwardEdge<boolean>[];
+): OntologyTypeVertexId[] =>
+  getOntologyTypesReferencedBy(
+    subgraph,
+    entityTypeId,
+    isConstrainsPropertiesOnEdge,
+  );
 
-  if (outwardEdges === undefined) {
-    return [];
-  }
-
-  return outwardEdges
-    .filter(isConstrainsPropertiesOnEdge)
-    .map((outwardEdge) => outwardEdge.rightEndpoint);
-};
+/**
+ * Gets identifiers for all `EntityType`s referenced within a given `EntityType` schema by searching for
+ * "InheritsFrom", "ConstrainsLinksOn" and "ConstrainsLinkDestinationsOn" `Edge`s from the respective `Vertex` within a `Subgraph`.
+ *
+ * @param subgraph {Subgraph} - The `Subgraph` containing the type tree of the `EntityType`
+ * @param entityTypeId {OntologyTypeVertexId | VersionedUrl} - The identifier of the `EntityType` to search for
+ * @returns {OntologyTypeVertexId[]} - The identifiers of the `EntityType`s referenced from the `EntityType`
+ */
+export const getEntityTypesReferencedByEntityType = (
+  subgraph: Subgraph<boolean>,
+  entityTypeId: OntologyTypeVertexId | VersionedUrl,
+): OntologyTypeVertexId[] =>
+  getOntologyTypesReferencedBy(
+    subgraph,
+    entityTypeId,
+    (outwardEdge): outwardEdge is StrictOntologyOutwardEdge =>
+      isInheritsFromEdge(outwardEdge) ||
+      isConstrainsLinksOnEdge(outwardEdge) ||
+      isConstrainsLinkDestinationsOnEdge(outwardEdge),
```

Comment:
Do we want these to be separated?

Reply:
Agreed, but talking about the return type of this function generally (not the predicate), if the caller only wants `ConstrainsLinksOn` edges, they'd need to refilter the output of this if we don't return separate collections.

(Splitting the _implementation_ up into 3 separate functions would end up with a traversal for each predicate as you say though so we probably shouldn't do that).

Reply:
Should we split the return value?

Reply:
Split them up in 09ccca724d78f531c83ffbf51bc6a647657a484e. The names/descriptions may need adjusting though

Reply:
But now to get all of the references we will require 3 traversals:
> We're not doing individual traversals for each predicate I guess

That's what I meant by this:
> (Splitting the implementation up into 3 separate functions would end up with a traversal for each predicate as you say though so we probably shouldn't do that).

I was suggesting we should split the _output_ like we have for the non-subgraph parallel implementation.

Alternatively we could have a method per edge kind, **and** a method to get all of the ontology ones (which is _not_ implemented as just calling all the methods per edge kind as that'd be triple the traversals)

Reply:
I completely misunderstood this, my bad. I see now that you explicitly said to split up the return values